framework/contrib/csvdiff/csvdiff

```diff
@@ -0,0 +1,386 @@
+        return self.__msg
+
+    def getNumErrors(self):
+        """Return number of errors"""
+        return self.__num_errors
+
+    def convertToTable(self, files):
+        """Convert text to a map of column names to column values"""
+        table_pair = []
+        for f in files:
+            f.seek(0)
+            text = f.read()
+            text = re.sub( r'\n\s*\n', '\n', text).strip()
+
+            # Exceptions occur if you try to parse a .e file
+            try:
+                lines = text.split('\n')
+                headers = lines.pop(0).split(',')
+                table = {}
+                for header in headers:
+                    table[header] = []
+
+                for row in lines:
+                    vals = row.split(',')
+                    if len(headers) != len(vals):
+                        self.addError(f, "Number of columns ("+str(len(vals))+") not the same as number of column names ("+str(len(headers))+") in row "+repr(row))
+                        break
+                    for header, val in zip(headers,vals):
+                        try:
+                            table[header].append(float(val))
+                        except:
+                            # ignore strings
+                            table[header].append(0)
+
+            except Exception as e:
+                self.addError(f, "Exception parsing file: "+str(e.args))
+                return {}
+
+            table_pair.append(table)
+
+        return table_pair
+
+    def parseComparisonFile(self, config_file):
+        """ Walk through comparison file and populate/return a dictionary as best we can """
+        # A set of known paramater naming conventions. The comparison file can have these set, and we will use them.
+        zero_params = set(['floor', 'abs_zero', 'absolute'])
+        tollerance_params = set(['relative', 'rel_tol'])
+
+        custom_params = {'RELATIVE' : 0.0, 'ZERO' : 0.0, 'FIELDS' : [] }
+        config_text = config_file.read()
+        line_settings = ''.join(re.findall(r"[^#]*", config_text)).split('\n')
+
+        for a_line in line_settings:
+            # Ignore TIME STEPS header (future csvdiff capability)
+            if a_line and a_line.lower().find('time steps') == 0:
+                continue
+
+            # This is the important line we were looking for, containing some default paramaters
+            if a_line and (set(a_line.split()).intersection(tollerance_params) or
+                           set(a_line.split()).intersection(zero_params)):
+                # populate important paramaters (floor, rel_tol, etc)
+                for param in set(a_line.split()).intersection(tollerance_params):
+                    custom_params['RELATIVE'] = re.findall(param + "\s+([0-9e.-]+)", a_line)[0]
+                for param in set(a_line.split()).intersection(zero_params):
+                    custom_params['ZERO'] = re.findall(param + "\s+([0-9e.-]+)", a_line)[0]
```

Comment:
Minor: You use set(a_line_split()) 4 times here. That's worth creating a variable so you don't have to create the set over and over.

Reply:
Also you have the same regex pattern in two places. You might just create the regex object once and use it twice. Helps with code maintainability too.